### PR TITLE
refactor(tsc): do not store some typescript declaration file text in multiple places

### PR DESCRIPTION
**Pull request by dsherret**
This stores some declaration files only in the tsc snapshot instead of in the snapshot and in the binary. Due to this, `deno types` now needs to get its information from a JS runtime.

Before: 69,987 KB (binary size on Windows)
After: ~~68,195 KB~~ 69,324 KB (in this PR I previously moved everything to the snapshot and it had a greater reduction in size probably because the snapshot is compressed, but that led to an increase in memory usage of 0.5MB)

### CODE REVIEW

File: cli/tsc/99_main_compiler.js [hunk 1]
```diff
@@ -385,7 +385,7 @@ delete Object.prototype.__proto__;
   // paths must be either relative or absolute. Since
   // analysis in Rust operates on fully resolved URLs,
   // it makes sense to use the same scheme here.
-  const ASSETS = "asset:///";
+  const ASSETS_URL_PREFIX = "asset:///";
```
Comment (bartlomieju) [inline cli/tsc/99_main_compiler.js:388]: We were supposed to change it to `deno:///assets/`, didn't we?

File: cli/tsc/mod.rs [hunk 2]
```diff
@@ -89,28 +68,57 @@ pub static COMPILER_SNAPSHOT: Lazy<Box<[u8]>> = Lazy::new(
...
+  let mut runtime = JsRuntime::new(RuntimeOptions {
+    startup_snapshot: Some(compiler_snapshot()),
+    extensions: vec![Extension::builder("deno_cli_tsc")
+      .ops(get_tsc_ops())
+      .build()],
+    ..Default::default()
+  });
+  let global =
+    runtime.execute_script("get_assets.js", "globalThis.getAssets()")?;
+  let scope = &mut runtime.handle_scope();
```
Comment (bartlomieju) [inline cli/tsc/mod.rs:119]: Hopefully `deno types` is not used by a hot path by any user code, this will incur about 20ms startup cost

**Review by bartlomieju — APPROVED** [review]

File: cli/tsc/99_main_compiler.js (hunk 1 above)
Comment (dsherret) [reply cli/tsc/99_main_compiler.js:388]: Not yet and I’m not sure yet. Need to investigate more

File: cli/tsc/mod.rs (hunk 2 above)
Comment (dsherret) [reply cli/tsc/mod.rs:119]: That sounds fine. People could cache it

**Pull request merged**